reconcile/utils/terraform_client.py

```diff
@@ -561,3 +575,37 @@ def split_to_lines(*outputs):
     def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
+
+    @staticmethod
+    def _is_ignored_rds_modification(resource_name: str,
+                                     resource_change: Mapping[str, Any],
+                                     output: Mapping[str, Any]) -> bool:
+        """
+        Determines whether the RDS resource changes are cases that we don't care about
+        """
+        before = resource_change['before']
+        after = resource_change['after']
+        changed_terraform_args = [
+            key for key, value in before.items() if value != after[key]
+        ]
+        if len(changed_terraform_args) == 1 \
+                and 'engine_version' in changed_terraform_args:
+            # TODO: I'm open to better ways to get creds. I looked at AWSApi, but I
+            #  wasn't convinced it was much better with the dependency on settings,
+            #  accounts, etc. that aren't accessible from this class
+            #  (without modification)
+            aws_config = output['configuration']['provider_config'][
+                'aws']['expressions']
+            access_key = aws_config['access_key']['constant_value']
+            secret_key = aws_config['secret_key']['constant_value']
+            rds_client = boto3.client('rds', aws_access_key_id=access_key,
+                                      aws_secret_access_key=secret_key)
```

Comment:
`self.accounts` has the accounts, `name` in this scope is the account name. so the only thing missing is really `settings`, which we can initiate the class with?

otherwise we'd start building a new aws_api within this class...

there will be more use cases in the future, so we might as well do it "right"? :)

one use case i can think of right now is to check if a parameter group is used when trying to delete it, for example.

Reply:
I ended up directly injecting `AWSApi` because that helps to avoid the need to `patch` to test this class. I didn't see any downside to this yet from initial testing. I still need to test the integrations that aren't `terraform_resources`.